File: emtk_asset/python/examples/rpk.py
import os
from pathlib import Path

from emtk_asset import Format
from emtk_asset.rpk import Rpk


def main() -> None:
    # Deserializing
    file_path = Path(os.environ["EXANIMA_EXE"]).parent.joinpath("Textures.rpk")
    print("Started deserializing from python")
    buf = file_path.read_bytes()
    format = Format.from_bytes(buf)
    print("Done deserializing from python")

    if isinstance(format._0, Rpk):
        rpk = format._0
        for entry in rpk.entries:
            print(entry.name)

    # Serializing
    file_path = file_path.parent.joinpath("Custom.rpk")
    print("Started serializing from python")
    buf = format.to_bytes()
    file_path.write_bytes(buf)
    print("Done serializing from python")


if __name__ == "__main__":
    main()

File: emtk_asset/examples/rpk.rs
use deku::prelude::*;
use emtk_asset::Format;
use std::{env, fs, io, path::PathBuf};

fn main() -> Result<(), Box<dyn std::error::Error>> {
	let mut file_path = PathBuf::from(env::var("EXANIMA_EXE")?);
	file_path.pop();
	file_path.push("Textures.rpk");

	// Deserializing
	let mut file = fs::File::open(&file_path)?;
	let mut buf_reader = io::BufReader::new(&mut file);
	let mut reader = Reader::new(&mut buf_reader);
	let format = Format::from_reader_with_ctx(&mut reader, ())?;

	// Serializing
	file_path.pop();
	file_path.push("Custom.rpk");
	let mut file = fs::File::create(&file_path)?;
	let mut buf_writer = io::BufWriter::new(&mut file);
	let mut writer = Writer::new(&mut buf_writer);
	format.to_writer(&mut writer, ())?;

	Ok(())
}
